build-tools/build.analyze.sh

```diff
@@ -6,7 +6,18 @@
 # export CCC_CC=clang-3.8
 # export CCC_CXX=clang++3.8
 
-BUILD_DIR=./build.analyze
+SCRIPT=$0
+while [ -h $SCRIPT ]
+do
+  BASE=$(dirname $SCRIPT)
+  SCRIPT=$(readlink $SCRIPT)
+  cd $BASE
+done
```

Comment:
What is this while loop for? (please add comment)

To get the absolute path of the current script, you can also do something like `base=$(cd $(dirname $SCRIPT) && pwd)`

Reply:
Sorry for the lack of documentation, will add it.

It is for the symlink in the rootdir, when calling `dirname` the path is relative to the symlink and not to the build-tools dir.

Reply:
It looks like the `--canonicalize` flag to `readlink` would do just that? (assuming that the while loop is meant to traverse the symlink chain down to the real file)

Reply:
Yes, but unfortunately this option is not available on all systems,
namely probably all BSD derivates.

But your right I'll put the loop in a function, with an appropriate name.